azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -284,7 +293,39 @@ public void cleanUpDanglingYarnApplications() {
     logger.info("Get the set of all executions: " +
         toBeCleanedContainers.stream().map(Object::toString).collect(Collectors.joining(",")));
 
+    if (executionStoppedFlows.isEmpty()) {
+      logger.info("No execution needs to kill yarn application, exit");
+      return;
+    }
+
     // TODO: For each of yarn clusters: get applications and kill those in the above set
+    for (Entry<String, Cluster> entry : this.allClusters.entrySet()) {
+      logger.info("clean up yarn applications in cluster:" + entry.getValue().clusterId);
+      cleanUpYarnApplicationsInCluster(toBeCleanedContainers, entry.getValue());
+    }
+  }
+
+  private void cleanUpYarnApplicationsInCluster(Set<Integer> toBeCleanedContainers,
+      Cluster cluster) {
+    org.apache.log4j.Logger apacheLogger =
```

Comment:
why not import this class to make the code clean?

Reply:
because the `Logger` class collides with existing `org.slf4j.Logger`